### DISCUSSION TITLE
Can I specify ports for zookeeper? In kind: Kafka

### DISCUSSION THREAD

**Discussion by simonebenati** [strimzi/strimzi-kafka-operator#8781] [Q&A] [answered]
Hello,

As the title says in the following yaml to configure kafka cluster is it possible to add a key in spec.zookeeper: to specify the port it needs to listen to? something similar to the listeners attribute for the brokers

```apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
    version: 2.7.0
    replicas: 1
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
      - name: tls
        port: 9093
        type: internal
        tls: true
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1
      log.message.format.version: "2.7"
      inter.broker.protocol.version: "2.7"
    storage:
      type: jbod
      volumes:
        - id: 0
          type: persistent-claim
          size: 100Gi
          deleteClaim: false
  zookeeper:
    replicas: 1
    storage:
      type: persistent-claim
      size: 100Gi
      deleteClaim: false
  entityOperator:
    topicOperator: { }
    userOperator: { }

**Comment by ppatierno** [comment]
It's not possible to do so. Strimzi configure ZooKeeper with the "usual and well known ports" from its ecosystem, so 2181 for clients connections, 3888 for quorum formation, 2888 for followers (to leader).
Why would you need to specify them?

**Reply by simonebenati** [reply]
Actually I misunderstood the task i was assigned.
We are trying to migrate from a legacy kafka installation to a kubernetes one and the developers, on the legacy installation would connect to zookeeper from the outside for some type of service discovery, at the moment strimzi kafka spawns a service of type clusterIP so it wouldn't be reachable from the outside the k8s cluster. Do you have any suggestions for this?

**Reply by ppatierno** [reply, marked as answer]
`for some type of service discovery` what do you mean by this? 
The ZooKeeper ensemble used by a Kafka cluster should be never used for anything else than ... storing metadata for the Kafka brokers. No applications should have access to it, but only the Kafka brokers. It could have a negative impact on the stability of the overall cluster.

**Reply by simonebenati** [reply]
Thank you for your answers, we decided that we'll deploy a zookeeper on the side of our kafka cluster deployed by strimzi to let the developers be able to do what they were doing prior.
I'm not able to answer your question as I'm not too competent yet and don't have the knowledge of what the developers are doing with kafka zookeeper.
But again, thank you for your support and help ! 🥇
